src/Renderer/ThreeExtended/PlanarControls.js

```diff
@@ -0,0 +1,746 @@
+/** Description: Camera controls adapted for a planar view, with animated movements
+* Left mouse button : "drag" the ground, translating the camera on the (xy) world plane.
+* Right mouse button : translate the camera on local x and world z axis (pan)
+* Ctrl + left mouse : rotate (orbit) around the camera's focus point.
+* Scroll wheel : zooms toward cursor position (animated).
+* Middle mouse button (wheel click) : 'smart zoom' at cursor location (animated).
+* S : go to start view (animated)
+* T : go to top view (animated)
+* How to use : instanciate PlanarControls after camera setup (setPosition and lookAt)
+* or you can also setup the camera with options.startPosition and options.startLook
+*/
+
+import * as THREE from 'three';
+
+// event keycode
+const keys = {
+    CTRL: 17,
+    SPACE: 32,
+    S: 83,
+    T: 84 };
+
+const mouseButtons = {
+    LEFTCLICK: THREE.MOUSE.LEFT,
+    MIDDLECLICK: THREE.MOUSE.MIDDLE,
+    RIGHTCLICK: THREE.MOUSE.RIGHT,
+};
+
+// control state
+const STATE = {
+    NONE: -1,
+    DRAG: 0,
+    PAN: 1,
+    ROTATE: 2,
+    TRAVEL: 3,
+};
+
+/**
+* PlanarControls Constructor
+* Numerical values have been adjusted for the example provided in examples/planar.html
+* Most of them can be changed with the options parameter
+* @param {PlanarView} view : the itowns view (planar view)
+* @param {options} options : optional parameters.
+*/
+function PlanarControls(view, options = {}) {
+    this.view = view;
+    this.camera = view.camera.camera3D;
+    this.domElement = view.mainLoop.gfxEngine.renderer.domElement;
+    this.position = this.camera.position;
```

Comment:
`this.position` isn't really useful: let's use `this.camera.position` directly instead.